src/omero_cli_render.py

```diff
@@ -537,6 +549,127 @@ def __info(self, args):
             finally:
                 img._closeRE()
 
+
+    def _get_datasets_for_image(self, gateway, img_id):
+        """
+        Get all datasets an image is part of.
+        :param gateway: Ref to gateway
+        :param img: The image
+        :return: The datasets
+        """
+        query = "select i from Image i join fetch i.datasetLinks idl " \
+                "join fetch idl.parent d where i.id = :id"
+        params = omero.sys.ParametersI()
+        params.map['id'] = rlong(img_id)
+        img = gateway.getQueryService().findByQuery(query, params)
+        datasets = []
+        for link in img.iterateDatasetLinks():
+            datasets.append(link.parent)
+        return datasets
+
+
+    @gateway_required
+    def impo(self, args):
+        """Implements the impo(rt) command"""
+        filename = args.channels
+        filename = filename.replace(".yml", "")
+        ds_name = filename.split(args.sep)[0]
+        img_name = filename.split(args.sep)[1]
+
+        set_args = args
+        if args.dataset:
+            ds_query = "select d from Dataset d where d.name = :name"
+            params = omero.sys.ParametersI()
+            params.map['name'] = rstring(ds_name)
+            datasets = self.gateway.getQueryService().findAllByQuery(ds_query, params)
+            if not datasets:
+                self.ctx.err(f"No dataset with name {ds_name} found.")
+                return
+            elif len(datasets) > 1:
+                self.ctx.dbg("More than one dataset found, using latest one")
+                target_ds = max(datasets, key=lambda ds: ds.getId().getValue())
```

Comment:
It looks like `target_ds` here is the Dataset ID, whereas `target_ds` below is the Dataset itself. Same for `target_img` below.

Reply:
Thanks. Looks I misunderstood this max function then.

Reply:
Ah - sorry, you might be right. I think I just misread this.